### DISCUSSION TITLE
Help: Support for Apple M Silicon?

### DISCUSSION THREAD

**Discussion by TimSusa** [savonet/liquidsoap#2132] [User Support]
**Is your feature request related to a problem? Please describe.**
I do not get it to run on apple m1 while trying to go according like here:

https://2020.vandragt.com/2016/traktor-streaming-to-shoutcast-servers-using-liquidsoap

- for me it does not work with brew
- for me it does not work with opam thingy
- I am just searching for the most simple solution getting artist info out of traktor handling it into butt streaming software, as it seems the most common use case arround 

Help would be appreciated as I am just trying to push open community with help aswell without any monetary aim.

**Comment by toots** [comment]
Hi!

I personally develop on a M machine so I'm pretty sure it works 🙂

The tap for liquidsoap is most likely outdated. `opam` should make it easier to install everything now. 

One thing to remember, though, is that `homebrew` on Apple silicon installs everything in a different location so you need to add the following to your environment, mostly via your shell rc file:
```
export CPATH=/opt/homebrew/include
export LIBRARY_PATH=/opt/homebrew/lib
```

After that, the following should get your pretty far ahead:
```
% brew install opam
% opam switch create 4.13.1
% opam install taglib mad ffmpeg lame vorbis cry samplerate ocurl liquidsoap
```

The full install instructions are here: https://www.liquidsoap.info/doc-dev/install.html#install-using-opam. I have omitted the `depext` step as most recent `opam` releases handle that automatically.
